### PR TITLE
Date picker improvements

### PR DESCRIPTION
**Pull request by collinjackson**

### CODE REVIEW

File: sky/packages/sky/lib/widgets/date_picker.dart [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 import 'dart:async';
 
+import 'package:sky/mojo/activity.dart' as activity;
```
Comment (abarth) [inline sky/packages/sky/lib/widgets/date_picker.dart]: I would just import this without the "as activity" part.

File: sky/packages/sky/lib/widgets/date_picker.dart [hunk 2]
```diff
@@ -299,12 +309,16 @@ class MonthPicker extends ScrollableWidgetList {
...
   DateTime _currentDate;
+  Timer _timer;
+
   void _updateCurrentDate() {
     _currentDate = new DateTime.now();
     DateTime tomorrow = new DateTime(_currentDate.year, _currentDate.month, _currentDate.day + 1);
     Duration timeUntilTomorrow = tomorrow.difference(_currentDate);
-    timeUntilTomorrow += const Duration(seconds: 1);  // so we don't miss it by rounding
-    new Timer(timeUntilTomorrow, () {
+    timeUntilTomorrow = const Duration(seconds: 1);  // so we don't miss it by rounding
```
Comment (abarth) [inline sky/packages/sky/lib/widgets/date_picker.dart]: Why = and not +=  ?

**Comment by eseidelGoogle** [comment]
lgtm

File: sky/packages/sky/lib/widgets/date_picker.dart [hunk 3]
```diff
@@ -43,12 +44,14 @@ class DatePicker extends StatefulComponent {
   DatePickerMode _mode = DatePickerMode.day;
 
   void _handleModeChanged(DatePickerMode mode) {
+    userFeedback.performHapticFeedback(activity.HapticFeedbackType_VIRTUAL_KEY);
```
Comment (abarth) [inline sky/packages/sky/lib/widgets/date_picker.dart]: s/activity.HapticFeedbackType_VIRTUAL_KEY/HapticFeedbackType_VIRTUAL_KEY/

File: sky/packages/sky/lib/widgets/date_picker.dart (hunk 2 above)
Comment (collinjackson) [inline sky/packages/sky/lib/widgets/date_picker.dart]: Good catch, I was testing the timer and forgot to change it back.

**Comment by abarth** [comment]
lgtm

**Pull request merged**